src/onnxruntime.cc

```diff
@@ -682,10 +682,40 @@ ModelState::AutoCompleteConfig()
   OrtAllocator* default_allocator;
   std::string model_path;
   {
+    TRITONSERVER_InstanceGroupKind kind;
+
+    triton::common::TritonJson::Value instance_group;
```

Comment:
From here to L713 should be wrapped inside TRITON_ENABLE_GPU. If it is a CPU only build, should always autofill kind to KIND_CPU

Reply:
The issue our users are seeing is that the onnxruntime_backend has been compiled with 'TRITON_ENABLE_GPU' and LoadModel() is getting passed a parameter which specifies to use the GPU when there is none. Wrapping this code in an `#ifdef` doesn't solve the problem since we still have to check the instance group kind. With that said, after looking at your suggested change, I agree that the code reads better for this section as it simplifies the logic of "autocomplete will only verify the instance kind if compiled with TRITON_ENABLE_GPU." If the backend is not compiled with the TRITON_ENABLE_GPU flag on, then don't worry about it.